docs/src/createEmotionCache.ts

```diff
@@ -8,8 +8,8 @@ function globalSelector(element: Element) {
   switch (element.type) {
     case RULESET:
       element.props = (element.props as string[]).map((value: any) => {
-        if (value.match(/(:where|:is)\(/)) {
-          value = value.replace(/\.[^:]+(:where|:is)/, '$1');
+        if (value.match(/(:where|:has)\(/)) {
+          value = value.replace(/\.[^:]+(:where|:has)/, '$1');
```

Comment:
It seems to break a bit the demo in https://deploy-preview-37362--material-ui.netlify.app/base/react-select/#introduction

Before:
<img width="367" alt="Screenshot 2023-05-22 at 17 01 46" src="https://github.com/mui/material-ui/assets/3165635/7a8f547f-8f81-405d-a09f-a0d2478eeca2">

After:
<img width="358" alt="Screenshot 2023-05-22 at 17 01 43" src="https://github.com/mui/material-ui/assets/3165635/3b170951-e4f8-4ca8-8bdd-35e7d7f032c3">

```suggestion
        if (value.match(/(:where)\(/)) {
          value = value.replace(/\.[^:]+(:where)/, '$1');
```

But it feels like we should remove this Emotion plugin. https://github.com/emotion-js/emotion/issues/2836#issuecomment-1406917978 feels more like a correct solution